components/Report/basic_usage.rst

```diff
@@ -0,0 +1,124 @@
+Basic Usage
+===========
+
+.. note::
+    First of all you need to register your data fetcher and renderer.
+    For more detailed information go to :ref:`Registry`_.
+
+.. _Registry: http://docs.sylius.org/en/latest/components/Registry/basic_usage.html
+
+.. _component_report_data-fetcher_delegating-data-fetcher:
+
+DelegatingDataFetcher
+---------------------
+
+This service allows you to delegate fetching to your custom registered service.
+Your data fetcher class should implement `DataFetcherInterface`_.
+
+.. _DataFetcherInterface: http://api.sylius.org/Sylius/Component/Report/DataFetcher/DataFetcherInterface.html
+
+.. code:: php
+
+    <?php
+
+    use Sylius\Component\Report\DataFetcher\DelegatingDataFetcher;
+    use Sylius\Component\Registry\ServiceRegistry;
+    use Sylius\Component\Report\Model\Report;
+    use Sylius\Component\Report\DataFetcher\DefaultDataFetchers;
+    use Sylius\Component\Report\Renderer\DefaultRenderers;
+    use Sylius\Component\Report\DataFetcher\DataFetcherInterface;
+    use Sylius\Component\Report\Renderer\RendererInterface;
+    use Sylius\Component\Report\Model\ReportInterface;
+    use Sylius\Component\Report\DataFetcher\Data;
+
+    /**
+     * Sales total data fetcher
+     */
+    class SalesTotalDataFetcher implements DataFetcherInterface
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function fetch(array $configuration)
+        {
+            // TODO: Implement fetch() method.
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getType()
+        {
+            // TODO: Implement getType() method.
+        }
+    }
+
+    $salesTotalDataFetcher = new SalesTotalDataFetcher();
+
+    $report = new Report();
+    $report->setDataFetcher(DefaultDataFetchers::SALES_TOTAL);
+    $report->setRenderer(DefaultRenderers::TABLE);
+
+    // Let's register our data fetcher.
+    $serviceRegistry = new ServiceRegistry(DataFetcherInterface::class);
+    $serviceRegistry->register(DefaultDataFetchers::SALES_TOTAL, $salesTotalDataFetcher);
```

Comment:
Use the constant here too. :) We should actually ditch these DefaultDataX classes. We should not assume anything in the component.